src/framework/application.js

```diff
@@ -1089,6 +1105,15 @@ Object.assign(Application.prototype, {
             this.i18n.assets = props.i18nAssets;
         }
 
+        if (props.areaLightData) {
+            asset = this.assets.get(props.areaLightData);
+            if (asset) {
+                this.setAreaLightData(asset);
+            } else {
+                this.assets.once('add:' + props.areaLightData, this.setAreaLightData, this);
+            }
+        }
+
```

Comment:
and even this should be a function to keep it cleaner

Reply:
moved to a function